docs/architecture/adr-080-libp2p-implementation.md

```diff
@@ -0,0 +1,157 @@
+# ADR 80: Libp2p Implementation Plan
+
+## Changelog
+
+- 2022-05-20: Initial Draft.
+
+## Status
+
+Accepted
+
+## Context
+
+Adopting libp2p is an larger project that is the culmination of a long
+programme of work in the tendermint codebase.
+
+## Alternative Approaches
+
+None. 
+
+The decision to adopt libp2p was part of ADR 73, and this ADR lays out
+an implementation plan and records a collection of decisions related
+to this implementation which have not been formally collected in any
+other form. 
+
+## Decisions
+
+### Feature Parity
+
+The goal of this project should be a p2p stack that has feature parity
+with the legacy stack. Everything that is possible with the p2p stack
+today should be possible in the future, eventually reducing the
+overhead of peer communication and complexity in the gossip/peer layer
+of the system. 
+
+In persuit of this goal, this project has two explicit *non-goals*:
+
+- Do not support message/wire compatibility with legacy stack.
```

Comment:
I think maybe you mean "do not _require_" here? We're still shipping around the same messages, at least at first?